### ISSUE TITLE
E3002: DurableConfig property not recognized on AWS::Serverless::Function

### ISSUE THREAD

**Issue by singledigit** [aws-cloudformation/cfn-lint#4395]
### CloudFormation Lint Version

1.42.0 (bundled with SAM CLI 1.150.1)

### What operating system are you using?

macOS 26.3

### Describe the bug

sam validate --lint reports E3002 for the DurableConfig property on AWS::Serverless::Function. DurableConfig is a valid SAM property for Lambda durable execution functions and deploys successfully with sam build, sam deploy, and sam sync. The cfn-lint schema for AWS::Serverless::Function doesn't include DurableConfig, so it's flagged as an unexpected additional property.

### Expected behavior

DurableConfig should be recognized as a valid property on AWS::Serverless::Function and not trigger E3002.

### Reproduction template

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31

Resources:
  MyDurableFunction:
    Type: AWS::Serverless::Function
    Properties:
      Runtime: nodejs20.x
      Handler: handler.handler
      CodeUri: ./src
      Timeout: 900
      AutoPublishAlias: live
      DurableConfig:
        ExecutionTimeout: 86400
        RetentionPeriodInDays: 3
```
